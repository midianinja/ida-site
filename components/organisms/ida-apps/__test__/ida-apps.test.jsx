import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import IDAApps from '../ida-apps';

describe('IDAApps', () => {
  afterEach(cleanup);
  
  it('renders without crashing', async () => {
    const { getByTestId } = render(<IDAApps />);
    expect(getByTestId('title')).toHaveTextContent('Onde você já pode usar o IDA?');
  });

  it('matches snapshot importance list', () => { 
    const tree = renderer.create(<IDAApps />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

