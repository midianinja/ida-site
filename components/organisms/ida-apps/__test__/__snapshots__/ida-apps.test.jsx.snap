// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IDAApps matches snapshot importance list 1`] = `
<section
  className="container"
>
  <div
    className="contentWrapper"
  >
    <h2
      className="title"
      data-testid="title"
    >
      Onde você já pode usar o IDA?
    </h2>
    <div
      className="cardsWrapper"
    >
      <div
        className="card"
        onClick={[Function]}
      >
        <img
          alt="SOM"
          data-testid="image"
          src="/static/images/som.png"
        />
        <h3
          className="title"
        >
          SOM
        </h3>
      </div>
      <div
        className="card"
        onClick={[Function]}
      >
        <img
          alt="500 Cidades"
          data-testid="image"
          src="/static/images/500cidades.png"
        />
        <h3
          className="title"
        >
          500 Cidades
        </h3>
      </div>
    </div>
    <div
      className="moreWrapper"
    >
      <img
        alt=""
        src="/static/icons/more.svg"
      />
      <p
        className="moreText"
      >
        Mais sites parceiros em breve, aguarde!
      </p>
    </div>
    <div
      className="buttonWrapper"
    >
      <button
        className="button __small"
        data-testid="button"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cadastre-se
      </button>
    </div>
  </div>
</section>
`;
